src/Core/ConstraintValidator/CustomerNameValidator.php

```diff
@@ -99,6 +99,11 @@ private function isPointSpacedValid($name)
         if (mb_strpos($name, '.') === false && mb_strpos($name, '。') === false) {
             return true;
         }
+
+        if (mb_strpos($name, '.') === 0 || mb_strpos($name, '。') === 0) {
+            return false;
+        }
+
         $pattern = $this->characterCleaner->cleanNonUnicodeSupport(self::PATTERN_DOT_SPACED);
```

Comment:
Only update `self::PATTERN_DOT_SPACES` by 
`(?!^)[\.。](\s{1}[^\ ]|$)`

Reply:
@PierreRambaud 

I thought about this but I think it is less efficient (in terms of performance) :
The minimal cases is to  : 
- Check if the name does not contain dots [see here](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/ConstraintValidator/CustomerNameValidator.php#L99)
- Check if the name contains at least a point at the beginning.

In these two cases it is useless to use `self::PATTERN_DOT_SPACES`

See https://we-love-php.blogspot.com/2014/03/strpos-vs-pregmatch-vs-stripos.html#:~:text=strpos()%20is%203%2D16,than%20using%20a%20long%20string

According to the official php docs for [preg_match](https://www.php.net/manual/fr/function.preg-match.php):

![image](https://user-images.githubusercontent.com/16455155/88203735-92289200-cc4a-11ea-8318-dbef36eb7f6e.png)

Reply:
I agree with preg_match, but here the regex is not only for checking if a char exists or not, it's searching for a position.
Using one preg_match is faster than two strpos + one preg_match :wink:

Reply:
See https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/ConstraintValidator/CustomerNameValidator.php#L102 and  https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/String/CharacterCleaner.php#L44

In the end you have well : `preg_replace` + `preg_match`

Reply:
I think the call of `$this->characterCleaner->cleanNonUnicodeSupport(` is also useless here, we control the content of the regex, it's not coming from a user :thinking: 
I think using one regex to do the job instead of multiple conditions to do one job should be better, but I'm not the only one who decide :sweat_smile:

Reply:
@PierreRambaud  done ! Thanks